### PR TITLE
Explicitly set agent to true

### PR DESCRIPTION
**Pull request by bernadinm**
This PR submitted by cmays20 explicitly sets the ssh agent to true. This PR enables CI/CD to confirm no regression.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -135,8 +135,9 @@ resource "null_resource" "instance-prereq" {
   count = "${coalesce(var.ami, var.user_data) == "" ? var.num : 0}"
 
   connection {
-    host = "${var.associate_public_ip_address ? element(aws_instance.instance.*.public_ip, count.index) : element(aws_instance.instance.*.private_ip, count.index)}"
-    user = "${module.dcos-tested-oses.user}"
+    host  = "${var.associate_public_ip_address ? element(aws_instance.instance.*.public_ip, count.index) : element(aws_instance.instance.*.private_ip, count.index)}"
+    user  = "${module.dcos-tested-oses.user}"
+    agent = true
```
Comment (sebbrandt87) [inline main.tf:140]: but this is for setting up a specific agent to use or to disable an agent usage.
per default `ssh-agent` is used, as on a windows machine we would need to set `pageant`.

**Review by sebbrandt87 — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (bernadinm) [reply main.tf:140]: Thanks @sebbrandt87 . I've also related the say with @cmays20 
 regarding the use for windows would need to set the pageant as listed in the terraform documentation. He did mention that by defaulting this value worked for him. I saw now risk in setting this by default as it doesn't affect the existing functionality.

https://www.terraform.io/docs/provisioners/connection.html#agent

The best solution here would be to allow users a flag that would control the `agent` flag from the top level module i.e [terraform-aws-dcos](https://github.com/dcos-terraform/terraform-aws-dcos) perpetuating the change to the submodules that would give the user the ability to set the value of `true` or `false` regardless of what operating system.  This change will simply make it explicitly `true` for the time being.

Comment (cmays20) [reply main.tf:140]: According to the docs, terraform only uses pageant on the windows side.  However, it won't use pageant unless agent is set to true (unlike on linux, which uses the ssh-agent without it being set explicitly to true).  I don't believe this needs to be bubbled up to the top unless you are going to expose all the variables possible for connections.  This change is safe though, as it really just fixes a bug on the terraform windows side.

Comment (sebbrandt87) [reply main.tf:140]: you are absolutely right, but this is not something read out of documentation but out of source code.
https://github.com/hashicorp/terraform/blob/929231a2e958372259f1daf95e243bc5ff9ff4bc/communicator/ssh/provisioner.go#L82-L89
So here is the real reason it needs to be set to true, as per default there is not going to be a env variable on the windows machine / shell pointing to a socket of pageant ... 

good to go with `agent = true`.

**Review by sebbrandt87 — APPROVED** [review]

File: Jenkinsfile [hunk 2]
```diff
@@ -1,39 +1,4 @@
 #!/usr/bin/env groovy
+@Library('jenkins-library@testing_integration') _
```
Comment (sebbrandt87) [inline Jenkinsfile]: please change the library pointing to the "master" for now, as we already merged the working `ci-deploy.sh` there.

File: ci-deploy.sh [hunk 3]
```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
```
Comment (sebbrandt87) [inline ci-deploy.sh]: remove this file as it is available within the jenkins-library already.

**Review by sebbrandt87 — CHANGES_REQUESTED** [review]

**Review by sebbrandt87 — APPROVED** [review]

**Pull request merged**